esp-hal-common/src/i2c.rs

```diff
@@ -0,0 +1,968 @@
+//! I2C Driver
+//!
+//! Supports multiple I2C peripheral instances
+use core::convert::TryInto;
+
+use embedded_hal::blocking::i2c::*;
+
+#[cfg(feature = "esp32c3")]
+use crate::pac::i2c::{RegisterBlock, COMD};
+#[cfg(not(feature = "esp32c3"))]
+use crate::pac::i2c0::{RegisterBlock, COMD};
```

Comment:
Is there any reason we shouldn't just rename `I2C` to `I2C0` for the ESP32-C3? Would be more consistent with this naming scheme and also simplify our code a bit.

Reply:
Good suggestion